src/engine/source/api/src/test/handlers.cpp

```diff
@@ -0,0 +1,412 @@
+#include "api/test/handlers.hpp"
+
+#include <api/adapter.hpp>
+#include <api/catalog/resource.hpp>
+
+#include <eMessages/eMessage.h>
+#include <eMessages/test.pb.h>
+
+#include "api/test/sessionManager.hpp"
+
+namespace
+{
+
+using namespace api::sessionManager;
+
+using std::shared_ptr;
+using std::string;
+
+using ::api::adapter::fromWazuhRequest;
+using ::api::adapter::genericError;
+using ::api::adapter::toWazuhResponse;
+
+} // namespace
+
+namespace api::test::handlers
+{
+
+namespace eEngine = ::com::wazuh::api::engine;
+namespace eTest = ::com::wazuh::api::engine::test;
+
+/**
+ * @brief Get the minimum available priority for a route.
+ *
+ * @param router Router instance.
+ * @return int32_t Minimum available priority. If no priority is available, returns -1.
+ */
+static inline int32_t getMinimumAvailablePriority(const shared_ptr<::router::Router>& router)
+{
+    // Create a set to store the taken priorities given the table
+    std::unordered_set<uint32_t> takenPriorities;
+
+    const auto routerTable = router->getRouteTable();
+    for (const auto& [name, priority, filter, policy] : routerTable)
+    {
+        takenPriorities.insert(priority);
+    }
+
+    int32_t minAvailablePriority {MINIMUM_PRIORITY};
+
+    // The condition may be confusing as the actual priority increases while the value decreases
+    while (takenPriorities.count(minAvailablePriority) > 0 && MAXIMUM_PRIORITY > minAvailablePriority)
+    {
+        // If priority is taken, decrease the value (so, increase the priority)
+        minAvailablePriority--;
+    }
+
+    return minAvailablePriority;
+}
+
+/**
+ * @brief Add a filter to the catalog.
+ *
+ * @param catalog Catalog instance.
+ * @param filterName Filter name.
+ * @param filterContent Filter content.
+ * @return std::optional<base::Error> If an error occurs, returns the error. Otherwise, returns std::nullopt.
+ */
+static inline std::optional<base::Error>
+addFilterToCatalog(shared_ptr<catalog::Catalog> catalog, const string& filterName, const string& filterContent)
+{
+    std::string error;
+
+    catalog::Resource targetResource;
+
+    try
+    {
+        targetResource = catalog::Resource {base::Name {"filter"}, catalog::Resource::Format::json};
+    }
+    catch (const std::exception& e)
+    {
+        error = e.what();
+    }
+
+    // If no error occurred, post the resource
+    if (error.empty())
+    {
+        const auto postResourceError = catalog->postResource(targetResource, filterContent);
+        if (postResourceError)
+        {
+            error = postResourceError.value().message;
+        }
+    }
+
+    return (error.empty() ? std::nullopt : std::make_optional(base::Error {error}));
```

Comment:
**thought:** Why calling `std::make_optional` here instead of just returning `base::Error(error)` ? If we had a more complex process to create the the base::Error object, it would definitely make more sense to use make_optional, but I don't think it's needed here because error is not constexpr and there are no narrowing conversions avoided either. std::optional is "smart" enough and it might improve readability. Thinking Out Loud. Please disregard is there is another valid reason to use make_optional.